platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/PathUtils.java

```diff
@@ -21,11 +23,104 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
 import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.MigrationReport;
+import org.codice.ddf.migration.MigrationWarning;
 
 public class PathUtils {
 
+  private static final String INVALID_NULL_PATH = "invalid null path";
+
+  /**
+   * Quietly deletes the specified file or recursively deletes the specified directory and audit the
+   * result. The provided directory is deleted.
+   *
+   * @param path the file or directory to be deleted
+   * @param type a string representing the type of file or directory to delete
+   * @return <code>true</code> if the file or directory was deleted, <code>false</code> otherwise
+   * @throws IllegalArgumentException if <code>path</code> or <code>type</code> is <code>null</code>
+   */
+  public static boolean deleteQuietly(Path path, String type) {
+    Validate.notNull(path, PathUtils.INVALID_NULL_PATH);
+    Validate.notNull(type, "invalid null type");
+    final File file = path.toFile();
+
+    if (FileUtils.deleteQuietly(file)) {
+      SecurityLogger.audit("{} file {} deleted", StringUtils.capitalize(type), file);
+      return true;
+    }
```

Comment:
:question: Do we need to audit on failure?

Reply:
Yes.